cql2pgjson/src/main/java/org/folio/cql2pgjson/util/Cql2SqlUtil.java

```diff
@@ -43,41 +55,38 @@ public static String cql2like(String s) {
         if (backslash) {
           like.append("\\\\");
           backslash = false;
-        } else {
-          backslash = true;
+          continue;
         }
         break;
       case '%':
       case '_':
-        like.append('\\').append(c);  // mask LIKE character
-        backslash = false;
+        if(!quotedString) {
+          like.append('\\').append(c);  // mask LIKE character
+        }
         break;
       case '?':
         if (backslash) {
           like.append("\\?");
-          backslash = false;
         } else {
           like.append('_');
         }
         break;
       case '*':
         if (backslash) {
           like.append("\\*");
-          backslash = false;
         } else {
           like.append('%');
         }
         break;
```

Comment:
If the CQL string contains an unescaped ? or an unescaped * then an exception should be thrown.